pkg/controllers/restore/controller.go

```diff
@@ -477,6 +474,31 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 	return nil
 }
 
+// customize provides customization of restored resource for edge cases
+func customize(obj *unstructured.Unstructured) {
+	switch obj.GetKind() {
+	case "ServiceAccount":
+		// remove secrets section as referenced secrets will be removed by k8s Token Controller as they are considered orphaned
+		delete(obj.Object, secretsMapKey)
+		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
+	case "Cluster":
+		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
+		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+			// only warn error if field can't be found. In case there are breaking api changes error will be printed as Warn and user has workaround to patch it.
+			redeployAgentGeneration, ok, err := unstructured.NestedInt64(obj.Object, "spec", "redeployAgentGeneration")
+			if err != nil {
+				logrus.Warnf("Fleet cluster %s/%s can't be reset to be re-imported, failed to fetch .spec.redeployAgentGeneration", obj.GetNamespace(), obj.GetName())
+				return
+			}
+			if ok {
```

Comment:
I think we need to handle `ok == false` case as well. On the object that I have on my environment this field is not set:

```
> kc get clusters.fleet.cattle.io -n fleet-default do-rke2-backup-test -o yaml                                                                        

apiVersion: fleet.cattle.io/v1alpha1
kind: Cluster
metadata:
  annotations:
...
spec:
  clientID: wzpsf4272bqbwln7tgcqpt2ttmftwj7fz9ps9zc95v9kdqgmkmsmnj
  kubeConfigSecret: do-rke2-backup-test-kubeconfig
status:
  agent:
...
```

Reply:
Good point. There is no need to check ok.